serverlessworkflow/modules/ROOT/pages/cloud/operator/using-persistence.adoc

```diff
@@ -29,82 +36,261 @@ spec:
+
+Additionally, for SonataFlow CR deployments that use the `preview` profile, it can configure the {product_name} build system to include specific Quarkus extensions required for persistence.
 
 [NOTE]
 ====
-Currently, PostgreSQL is the only persistence supported.
+Currently, PostgreSQL is the only supported persistence.
 ====
 
-* Using the custom defined persistence in the `SonataFlow` CR
+To learn how to initialize the database schema see: <<_database_schema_initialization, Database schema initialization>>.
 
-Alternatively, you can define a dedicated configuration in the `SonataFlow` CR instance using the same schema format found in the Platform CRD:
+== Persistence configuration precedence rules
 
-[source,yaml,subs="attributes+"]
+<<_configuring_the_persistence_using_the_sonataflow_cr, SonataFlow CR persistence>> can be used with or without the <<_configuring_the_persistence_using_the_sonataflowplatform_cr, SonataFlowPlatform CR persistence>>.
+
+And, if the current namespace has an already configured <<_configuring_the_persistence_using_the_sonataflowplatform_cr, SonataFlowPlatform CR persistence>>, the following rules apply:
+
+* If the SonataFlow CR has a configured persistence, that configuration will apply.
+* If the SonataFlow CR has no configured persistence, i.e., the field `spec.persistence` is not present at all, the persistence configuration will be taken from the current platform.
+* If you don't want the current workflow to use persistence, you must use the following configuration in the SonataFlow CR: `spec.persistence : {}` to ban the SonataFlowPlatform persistence.
+
+== Persistence configuration and SonataFlow profiles
+
+All the configurations shown in this guide apply exactly the same for both the `preview` and the `gitops` profiles, however, you must not use them when you use the `dev` profile.
+
+Finally, the only distinction between `preview` and `gitops` profiles, is that when you use the `gitops` profile, the following Quarkus extensions must be added when you build your workflow image.
+
+[cols="40%,40%,20%", options="header"]
+|===
+|groupId
+|artifactId
+|version
+
+| {groupId_quarkus-agroal}
+| {artifactId_quarkus-agroal}
+| {quarkus_version}
+
+| {groupId_quarkus-jdbc-postgresql}
+| {artifactId_quarkus-jdbc-postgresql}
+| {quarkus_version}
+
+| {groupId_kie-addons-quarkus-persistence-jdbc}
+| {artifactId_kie-addons-quarkus-persistence-jdbc}
+| {kogito_version}
+|===
+
+This last can be done by passing the following build argument to the `kogito-swf-builder`:
+
+
+[source,bash,subs="attributes+"]
+----
+QUARKUS_EXTENSIONS={groupId_quarkus-agroal}:{artifactId_quarkus-agroal}:{quarkus_version},{groupId_quarkus-jdbc-postgresql}:{artifactId_quarkus-jdbc-postgresql}:{quarkus_version},{groupId_kie-addons-quarkus-persistence-jdbc}:{artifactId_kie-addons-quarkus-persistence-jdbc}:{kogito_version}
```

Comment:
Can't we configure the platform `.spec.build.template` with this information? Or even better, don't we already add these extensions in the `preview` profile? Maybe redirecting the user to the guide that teaches how to modify the builder image?

Reply:
@ricardozanini  yes, in the `preview`  we already ad them, so the guide here tries to say.

For `dev`never include.

For `preview`, do nothing. (we automatically inject these  extensions at build time, this is the development I did)

For `gitops` you must add them by your self whey you build the WF, because we can't handle the image build in that case.

Not sure if this clarifies?